src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_DATA = {};
+    private static final int ZERO = 0;
+    private static final int ONE = 1;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = ZERO;
+    }
 
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            grow(size + ONE);
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        if (elements.length == size) {
+            grow(size + ONE);
+        }
+        System.arraycopy(elements, index, elements, index + ONE, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] listElements = (T[]) new Object[list.size()];
+        for (int i = ZERO; i < list.size(); i++) {
+            listElements[i] = list.get(i);
+        }
+        System.arraycopy(listElements, ZERO, elements, size, listElements.length);
```

Comment:
```suggestion
        System.arraycopy(listElements, 0, elements, size, listElements.length);
```
не переусердствуй з magic numbers)

Reply:
Понял, просто уже в привычку вошло использовать их везде 😄